homeassistant/components/nmbs/config_flow.py

```diff
@@ -122,25 +123,20 @@ async def async_step_import(self, user_input: dict[str, Any]) -> ConfigFlowResul
         except CannotConnect:
             return self.async_abort(reason="api_unavailable")
 
-        station_from = None
-        station_to = None
-        station_live = None
-        for station in self.stations:
-            if user_input[CONF_STATION_FROM] in (
-                station["standardname"],
-                station["name"],
-            ):
-                station_from = station
-            if user_input[CONF_STATION_TO] in (
-                station["standardname"],
-                station["name"],
-            ):
-                station_to = station
-            if CONF_STATION_LIVE in user_input and user_input[CONF_STATION_LIVE] in (
-                station["standardname"],
-                station["name"],
-            ):
-                station_live = station
+        # Find stations with fallback
+        station_from = await async_find_station_with_fallback(
+            self.hass, user_input[CONF_STATION_FROM], self.api_client
```

Comment:
So, the tests break when this runs, is it mostly because the test framework hasn't set up `hass.data[DOMAIN]` properly?